report_balance_sheet.php

```diff
@@ -0,0 +1,168 @@
+                        <th></th>
+                        <th class="text-uppercase">Total Liabilities</th>
+                        <th class="text-right"><?php echo $formatted_total_liabilities; ?></th>
+                    </tr>
+
+                    <!-- Equity Section -->
+                    <tr>
+                        <th colspan="3" >Equity</th>
+                    </tr>
+                    <?php
+                    mysqli_data_seek($result_accounts, 0); // Reset the result pointer to the start
+                    while ($row = mysqli_fetch_array($result_accounts)) {
+                        $balance = $row['opening_balance'] + $row['total_payments'] + $row['total_revenues'] - $row['total_expenses'];
+                        $account_type = $row['account_type'];     
+                        if ($account_type >= 30) {
+                            // Display equity account row
+                            echoAccountRow($row, $balance);
+                            $total_equity += $balance;
+                            $formatted_total_equity = numfmt_format_currency($currency_format, $total_equity, $row['account_currency_code']);
+                        }
+                    }
+                    ?>
+                    <tr>
+                        <th></th>
+                        <th class="text-uppercase">Total Equity</th>
+                        <th class="text-right"><?php echo $formatted_total_equity; ?></th>
+                    </tr>
+                    <!-- Total Equity and Liabilities -->
+
+                    <?php
+                        $total_liabilities_and_equity = $total_liabilities + $total_equity;
+                        $formatted_total_liabilities_and_equity = numfmt_format_currency($currency_format, $total_liabilities_and_equity, $currency);
+                    ?>
+
+                    <tr>
+                        <th></th>
+                        <th class="text-uppercase">Total Liabilities and Equity</th>
+                        <th class="text-right"><?php echo $formatted_total_liabilities_and_equity; ?></th>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+    </div>
+</div>
+
+<?php require_once("footer.php"); ?>
+
+<?php
+function echoAccountRow($accountRow, $balance) {
+    global $currency_format;
+    $account_type_strings = [
+        11 => "Current Assets",
+        12 => "Fixed Assets",
+        13 => "Other Assets",
+        21 => "Current Liabilities",
+        22 => "Long Term Liabilities",
+        23 => "Other Liabilities",
+        30 => "Equity"
+    ];
+    $account_type_string = $account_type_strings[$accountRow['account_type']] ?? "Unknown";
+    $account_name_encoded = urlencode($accountRow['account_name']);
+    echo "
+    <tr>
+        <td>$account_type_string</td>
+        <td><a class=\"text-dark\" href=\"account_details.php?account_name=$account_name_encoded\">{$accountRow['account_name']}</a></td>
```

Comment:
- All non-int variables should be echoed/assigned using nullable_htmlentities to prevent XSS

- Ideally (where possible) we try to assign the variable once (e.g. as $account_type and $account_name and then just re-use as required), rather than constantly reusing $accountRow['account_type'] - its much cleaner

Reply:
This has been resolved. Looking over the rest of the code now.